GeoLib/StationBorehole.cpp

```diff
@@ -55,7 +55,7 @@ int StationBorehole::find(const std::string &str)
     std::size_t size = _soilName.size();
     for (std::size_t i = 0; i < size; i++)
     {
-        if (_soilName[i].find(str) == 0)
+        if (_soilName[i].compare(str) == 0)
```

Comment:
Isn't it simply same as `_soilName[i] == str`?
And then the whole StationBorehole::find() is simply a
```c++
std::find_first_of(begin(_soilName), end(_soilName), str) != _soilName.end()
```